pkg/executor/core/executor.go

```diff
@@ -62,87 +73,114 @@ func (executor Executor) VerifySubject(ctx context.Context, verifyParameters e.V
 	return result, nil
 }
 
-func (executor Executor) GetVerifyRequestTimeout() time.Duration {
-	timeoutMilliSeconds := defaultVerifyRequestTimeoutMilliseconds
-	if executor.Config != nil && executor.Config.VerificationRequestTimeout != nil {
-		timeoutMilliSeconds = *executor.Config.VerificationRequestTimeout
+// verifySubjectForRegoPolicy verifies the subject with results.
+// And it also returns the decision based on the verifier results if required.
+func (executor Executor) verifySubjectForRegoPolicy(ctx context.Context, verifyParameters e.VerifyParameters) (types.VerifyResult, error) {
+	results, err := executor.verifySubjectInternalWithoutDecision(ctx, verifyParameters)
+	if err != nil {
+		return types.VerifyResult{}, err
 	}
-	return time.Duration(timeoutMilliSeconds) * time.Millisecond
+	// If it requires embedded Rego Policy Engine make the decision, execute
+	// OverallVerifyResult to evaluate the overall result based on the policy.
+	result := types.VerifyResult{VerifierReports: results}
+	if !featureflag.PassthroughMode.Enabled {
+		result.IsSuccess = executor.PolicyEnforcer.OverallVerifyResult(ctx, result.VerifierReports)
+	}
+
+	return result, nil
 }
 
-func (executor Executor) GetMutationRequestTimeout() time.Duration {
-	timeoutMilliSeconds := defaultMutateRequestTimeoutMilliseconds
-	if executor.Config != nil && executor.Config.MutationRequestTimeout != nil {
-		timeoutMilliSeconds = *executor.Config.MutationRequestTimeout
+// verifySubjectInternalWithDecision verifies the subject and makes the decision
+// based on the policy. It is only used internally for the Json-based policy.
+func (executor Executor) verifySubjectInternalWithDecision(ctx context.Context, verifyParameters e.VerifyParameters) (types.VerifyResult, error) {
+	verifierReports, err := executor.verifySubjectInternalWithoutDecision(ctx, verifyParameters)
+	if err != nil {
+		return types.VerifyResult{}, err
 	}
-	return time.Duration(timeoutMilliSeconds) * time.Millisecond
+	if len(verifierReports) == 0 {
+		return types.VerifyResult{}, ErrReferrersNotFound
```

Comment:
Just a note there, sometime the issue could be "no verifiers was able to validate the artifact" , the ErrReferrersNotFound maybe not be the best default error. But i think this is the existing code path, we should updated this logic when we get a chance to.

Reply:
Synced offline. I'll keep the implementation, but add a version to the verificationResult and add more docs explaining it.